### PR TITLE
Enable NVM for Thread devices

### PR DESCRIPTION
**Pull request by artokin**
Allow Thread devices to store content to NVM.

### CODE REVIEW

File: configs/mesh_thread.json [hunk 1]
```diff
@@ -11,20 +11,24 @@
         "enable-led-control-example": true,
         "LED": "NC",
         "BUTTON": "NC"
-    },  
+    },
+    "macros": ["MBEDTLS_USER_CONFIG_FILE=\"mbedtls_config.h\""],
```
Comment (SeppoTakalo) [inline configs/mesh_thread.json]: Are you sure that this builds on all the platforms that we are targeting in our CI?

File: mesh_nvm.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#define TRACE_GROUP "anvm"
+
+#if MBED_CONF_APP_MESH_TYPE == MESH_THREAD
+LittleFileSystem fs("fs");
+BlockDevice* bd = storage_selector();
+
+void mesh_nvm_initialize() 
+{
+    fflush(stdout);
```
Comment (SeppoTakalo) [inline mesh_nvm.cpp]: Why `fflush()`?

File: mesh_nvm.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    int err = fs.mount(bd);
+    if (err) {
+        tr_warning("mount error: %d, trying format...", err);
+        fflush(stdout);
```
Comment (SeppoTakalo) [inline mesh_nvm.cpp]: Again, why `fflush()`?

File: mesh_nvm.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#include "mbed_trace.h"
+#include "storage-selector.h"
+#include "LittleFileSystem.h"
+#include "ns_file_system.h"
+
+#define TRACE_GROUP "anvm"
+
+#if MBED_CONF_APP_MESH_TYPE == MESH_THREAD
+LittleFileSystem fs("fs");
+BlockDevice* bd = storage_selector();
```
Comment (SeppoTakalo) [inline mesh_nvm.cpp]: How does this work? Is this reliable to be called before `main()` or should you initialize `*bd` only at the `mesh_nvm_initialize()`

**Comment by artokin** [comment]
@SeppoTakalo , @mikter , @karsev would you please review?

**Comment by karsev** [comment]
Could we have mesh_nvm.cpp under the mesh-api? Would be then easy to enable nvm from other apps.

**Comment by SeppoTakalo** [comment]
@karsev That is a good proposal, but Mbed OS cannot have any external dependencies and NVM depends on SD card driver.

Once the storage selector and SD drivers enter the Mbed OS root, be can move those to mesh-API.

**Pull request merged**